app/(nms)/network-configuration/modals.tsx

```diff
@@ -0,0 +1,446 @@
+import { apiGetAllDeviceGroupNames } from "@/utils/callDeviceGroupApi";
+import { Button, Form, Input, ConfirmationButton, Modal, Select } from "@canonical/react-components"
+import { createNetworkSlice, editNetworkSlice, deleteNetworkSlice } from "@/utils/networkSliceOperations";
+import { getGnbList } from "@/utils/gnbOperations";
+import { getUpfList } from "@/utils/upfOperations";
+import { GnbItem, NetworkSlice, UpfItem } from "@/components/types";
+import { queryKeys } from "@/utils/queryKeys";
+import { useAuth } from "@/utils/auth"
+import { useFormik } from "formik";
+import { useQuery, useQueryClient } from "@tanstack/react-query"
+import { useState } from "react"
+import { OperationError, is401UnauthorizedError}  from "@/utils/errors";
+
+import * as Yup from "yup";
+import ErrorNotification from "@/components/ErrorNotification";
+
+
+interface NetworkSliceFormValues {
+  name: string;
+  mcc: string;
+  mnc: string;
+  sst: string;
+  upf: UpfItem;
+  gnbList: GnbItem[];
+}
+
+const NetworkSliceSchema = Yup.object().shape({
+    name: Yup.string()
+      .min(1)
+      .max(20, "Name must not exceed 20 characters")
+      .matches(/^[a-zA-Z0-9-_]+$/, {
```

Comment:
The first character will be allowed to be a dash or an underscore, I would suggest forcing the first character to be a letter, like so:
```suggestion
      .matches(/^[a-zA-Z][a-zA-Z0-9-_]+$/, {
```

Reply:
done, we need to fix this in the webconsole as well